### ISSUE TITLE
$user_level contains a CRLF after altlogin.php is used

### ISSUE THREAD

**Issue by AlienXAXS** [3RaGaming/Web_Control#41]
Hi,
$user_level contains a \r\n after attempting to login via altlogin.php, i've not tried discord yet.
Because of this, the Javascript that is created in index.php @ line 49 is faulty.

I have fixed this myself by adding a trim() around the altlogin.php variable $userL on line 44

**Comment by zackman0010** [comment]
I believe @stuudmuffin fixed that one in one of his recent commits. Can you confirm?

**Comment by AlienXAXS** [comment]
Hey,
Does your installer pull the latest commit from the repo? I only ask as I installed it from the bootstrapper after his latest commit.

I am however using CentOS

**Comment by zackman0010** [comment]
Yes, the installer pulls directly from the master branch here.
I do remember CentOS having some additional problems with working. Apache's default user in CentOS isn't `www-data` like Ubuntu (and therefore Web Control) are set up with. There is a way to change this, but that's something @stuudmuffin did when we tested. 
As for the newline issue, Stud said he thought he fixed it. He'll take another look when he wakes up, he was kind of up all night working on those commits. Lol

**Comment by zackman0010** [comment]
Your custom fix should work for now, however.

**Comment by stuudmuffin** [comment]
I did realize something was off with this. I'm surprised we haven't caught it sooner, since I think one of our admin's only uses the alt login. I had not had time to look through it yet. 

Need to combine the altlogin and normal login in a way to support using each through the same page.

**Comment by stuudmuffin** [comment]
Alien, using Centos may have some difficultly. Firstly, depending on your version of centos, you may need to manually install php7, as I believe centos still uses php5 by default.

Also, the httpd/nginx used by centos uses "apache" as the web user instead of www-data. This complicates things as much of our script assumes the apache user is going to be "www-data". 
Running centos, I would recommend editing the httpd.conf file to start the web server under the www-data user. This user must also be created
`useradd www-data`
`groupadd www-data`
you can then search for files owned by apache using:
`find / -user apache`
change all permissions for files owned by apache using
`chown -R www-data:www-data DIRECTORY_PATH`
using -R will recursively change permissions for all files in a directory.

Anytime you `yum update` your system, if php is ever updated, you will need to run these commands again.

With the advent of our automated install, we may be able to forego hard-setting the www-data user. But this is all dependent on a system admin remembering to change file permissions in /var/www for the apache service user. Any files not owned by the web server are likely to reduce functionality of the web control, and are difficult to track down if one is not looking for this issue. Hence, our safeguard to force files to use a specific user, so we can cronjob ensuring they are owned properly.

**Comment by AlienXAXS** [comment]
Hey @stuudmuffin, thanks for your reply.
I will try your software again soon and run it on ubuntu next time, I am just a CentOS man myself so it's my go-to distro.

As for the perms, yeah i did change the running user but did not think about the implications of running a yum update on the system overwriting my changes, good call on that.

As for the hard coded username www-data, is it not possible to perhaps read the distro, then find the apache httpd.conf to read the user from there in a future release to automatically decide what user the software should be using?

**Comment by stuudmuffin** [comment]
It could be done; setting the user dynamically. We have just not gotten to that yet. We need a more centralized config location for the web control, then it can be easily set which distro is being used, so the script will know how to handle certain scenarios.

This is a change that will come in time.

**Comment by AlienXAXS** [comment]
Hey,
I have installed an Ubuntu VM now and got the software installed - the installer seemed to work fine and i can altlogin to the utility now, however I am unable to start any servers.

I have modified my sudoers file as requested by the requirements section but the server instance just doesnt start.

Log files just show: "Starting Server. Load Latest. Initiated by alienx"

** Edit **
In my screenlog file (under /var/www/factorio/logs) i have this:
`server1$start$true,34291,/var/www/factorio/server1,/usr/share/factorio/0.15.13/bin/x64/factorio
server1$status
`
